vm/compiler/src/coinbase_puzzle/tests.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use console::{account::*, network::Testnet3};
+use snarkvm_utilities::Uniform;
+
+use rand::RngCore;
+
+#[test]
+fn test_coinbase_puzzle() {
+    let max_degree = 1 << 15;
+    let mut rng = rand::thread_rng();
```

Comment:
```suggestion
    let mut rng = snarkvm_utilities::test_rng();
```
or `TestRng` if https://github.com/AleoHQ/snarkVM/pull/1013 is merged first